pinot-common/src/test/java/com/linkedin/pinot/common/utils/TarGzCompressionUtilsTest.java

```diff
@@ -138,26 +136,24 @@ public void testBadFilePath() throws Exception {
     Assert.assertFalse(metaFile.isDirectory());
 
     File tarGzPath = new File(tarDir, SEGMENT_NAME + ".tar.gz");
-    mockIncorrectData(metaFile, tarGzPath);
+    createInvalidTarFile(metaFile, tarGzPath);
 
     try {
       TarGzCompressionUtils.unTar(tarGzPath, untarDir);
       Assert.fail("Did not get exception!!");
-    } catch (IOException e) {
-      Assert.assertTrue(e.getMessage().contains(TarGzCompressionUtils.EXTRACT_FILE_OUTSIDE_OF_TARGET_DIR));
+    } catch (Exception e) {
+      Assert.assertTrue(e instanceof IOException);
+      Assert.assertTrue(e.getMessage().startsWith("Tar file must not"));
```

Comment:
thanks for adding this string check.  As there are so many exceptions possible, it is best to check that we are getting the exception that we expect, from the place we expect it.
An alternative is to use a specific exception (in this case, a new pinot exception perhaps), but I am fine with this.